#! /bin/sh -e

## DP: fixes missing get_mountpoint() error by statically linking in the 
## mountpoint code.
## DP: Author: Jaldhar H. Vyas <jaldhar@debian.org>

. $(dirname $0)/DPATCH

exit 0
@DPATCH@
diff -ruN dovecot-1.0.rc2.orig/src/plugins/quota/Makefile.am dovecot-1.0.rc2/src/plugins/quota/Makefile.am
--- dovecot-1.0.rc2.orig/src/plugins/quota/Makefile.am	2006-06-17 14:02:55.000000000 -0400
+++ dovecot-1.0.rc2/src/plugins/quota/Makefile.am	2006-07-06 16:21:25.000000000 -0400
@@ -12,6 +12,11 @@
 module_LTLIBRARIES = \
 	lib01_quota_plugin.la
 
+# get some functions included which only plugins use. liblib should probably
+# be a shared library so this wouldn't be needed..a
+unused_objects = \
+	../../lib/mountpoint.o
+
 lib01_quota_plugin_la_SOURCES = \
 	quota.c \
 	quota-fs.c \
@@ -21,6 +26,8 @@
         quota-plugin.c \
 	quota-storage.c
 
+lib01_quota_plugin_la_LIBADD = $(unused_objects)
+
 noinst_HEADERS = \
 	quota.h \
 	quota-fs.h \
